src/sensai/util/geometry.py

```diff
@@ -89,3 +91,64 @@ def add_edge(edge_index_pair, edge_vertex_pair):
     remaining_edges = MultiLineString(edge_vertex_pairs)
 
     return unary_union(list(polygonize(remaining_edges)))
+
+
+class SpanningTree:
```

Comment:
Since a spanning tree is a concept from graph theory (not geometry), users might not go looking for something like this here.

Reply:
Yeah, I thought the same. In my original repo I had it in a module called trees. When migrating, I didn't want sensAI to become too much of a graph/geometry oriented library, so I hesitated to create a new module. Do you have a suggestion for where to put it?

Reply:
Since graph theory is essential to many AI algorithms, a package dealing with graph-related stuff would not be out of place.

Reply:
One thing to consider: It could look odd to someone reading the code, as it's not a common pattern to only conditionally set an attribute. (Also: no docstring explaining it.) That's why I opted for the variant that makes the fallback to the class attribute explicit.

Reply:
cool, I'll move it to util.graph